File: app/src/main/java/wangdaye/com/geometricweather/db/converters/TimeZoneConverter.java
package wangdaye.com.geometricweather.db.converters;

import androidx.room.TypeConverter;

import java.util.TimeZone;

public class TimeZoneConverter {

    @TypeConverter
    public TimeZone convertToEntityProperty(String databaseValue) {
        if (databaseValue == null) {
            return null;
        }
        return TimeZone.getTimeZone(databaseValue);
    }

    @TypeConverter
    public String convertToDatabaseValue(TimeZone entityProperty) {
        if (entityProperty == null) {
            return null;
        }
        return entityProperty.getID();
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/db/converters/TimeZoneConverter.kt
package wangdaye.com.geometricweather.db.converters

import androidx.room.TypeConverter
import java.util.TimeZone

class TimeZoneConverter {

    @TypeConverter
    fun convertToEntityProperty(databaseValue: String?): TimeZone? {
        if (databaseValue == null) {
            return null
        }
        return TimeZone.getTimeZone(databaseValue)
    }

    @TypeConverter
    fun convertToDatabaseValue(entityProperty: TimeZone?): String? {
        return entityProperty?.id
    }
}
